Add list of files to report of 'wp doctor check cache-flush' command

kiranpotphode opened the pull request:
Fixes #184

This PR adds enhancement for the `wp doctor check cache-flush` command as mentioned in issue #184

The output will show the list of files where `wp_cache_flush()` is used.

wojtekn commented on inc/checks/class-cache-flush.php:
```diff
@@ -36,6 +32,6 @@ public function run() {
 
 		$this->set_status( 'warning' );
-		$this->set_message( 'Use of wp_cache_flush() detected.' );
+		$this->set_message( 'Use of wp_cache_flush() detected in ' . implode( ', ', $this->_matches ) );
```
What do you think about improving the output readability, e.g., by removing the site path from the matches to display relative paths?

kiranpotphode replied:
I have now modified the code to include relative paths only.

wojtekn commented:
@kiranpotphode, thanks for adjusting the output format, it's now more readable.

I've just checked tests and found out we will need to update those as they rely on the old message format:

```
composer behat -- features/check-cache-flush.feature
```

kiranpotphode commented:
> @kiranpotphode , thanks for adjusting the output format, it's now more readable.
> 
> I've just checked tests and found out we will need to update those as they rely on the old message format:
> 
> ```
> composer behat -- features/check-cache-flush.feature
> ```

@wojtekn  Thanks for pointing that out. 
I have updated the test now. I also updated the doc block comment for the class.

wojtekn reviewed (approved):
Looks good, thanks for updating the test!

ernilambar commented:
@kiranpotphode Can you please rebase the PR with main branch? Now `behat.yml` file has been added. So GitHub actions will run with all functional tests.

kiranpotphode commented:
> @kiranpotphode (https://profiles.wordpress.org/kiranpotphode) Can you please rebase the PR with main branch? Now `behat.yml` file has been added. So GitHub actions will run with all functional tests.

@ernilambar 
I have updated my branch with the latest code from the main branch.
However, I noticed that some tests are failing on GitHub actions.

But my tests on local are passing for the following command.
`composer behat -- features/check-cache-flush.feature`

Can you guide what might the problem?

ernilambar commented:
Yah, I noticed. `wp_cache_flush()` is there in `sqlite-database-integration` which is used for Sqlite implementation. Probably you have tested with `mysql` in your setup. That is why you are not seeing error in local.

kiranpotphode commented:
I have updated the tests. Now `wp_cache_flush()` related issues are fixed.

Still noticing some other failed tests for `WP trunk on PHP 8.4 ` Scenario though.

ernilambar commented:
@kiranpotphode WP is still not compatible to PHP `8.4`. So we can ignore that now.
